config/software/datadog-agent-integrations.rb

```diff
@@ -50,45 +52,43 @@
       conf_directory = "../../extra_package_files/EXAMPLECONFSLOCATION"
     end
 
-    all_reqs_file = File.open("#{project_dir}/check_requirements.txt", 'w+')
-    # Manually add "core" dependencies that are not listed in the checks requirements
-    # FIX THIS these dependencies have to be grabbed from somewhere
-    all_reqs_file.puts "wheel==0.30.0 --hash=sha256:e721e53864f084f956f40f96124a74da0631ac13fbbd1ba99e8e2b5e9cafdf64"\
-        " --hash=sha256:9515fe0a94e823fd90b08d22de45d7bde57c90edce705b22f5e1ecf7e1b653c8"
-
-    all_reqs_file.close
-
-    # Install all the requirements
-    if windows?
-      pip "install -r #{project_dir}/check_requirements.txt"
-    else
-      build_env = {
-        "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
-        "PATH" => "#{install_dir}/embedded/bin:#{ENV['PATH']}",
-      }
-      pip "install -r #{project_dir}/check_requirements.txt", :env => build_env
-    end
-
-    # Set frozen requirements
-    pip "freeze > #{install_dir}/agent_requirements.txt"
```

Comment:
We still need to build `agent_requirements` even if we no longer explicitly install the `wheel` package. We need to create this *after* installing pip tools.

The `"{windows_safe_path(project_dir)}\\datadog_checks_base\\agent_requirements.in` isn't the same thing we need the constraints file to ensure the omnibus deps for agent5 are respected above all things. The problem this has had for a long time is that pip will enforce the constraints but will not fail installing a package that doesn't meet the requirements. I know that sucks, ideally we'd want to fail the build :(

Reply:
Ok thats good to know. Thanks for the info! I put back the freeze and the use of it as a constraints file. I agree it would be better to fail the build but there isn't a way to do that currently.